### PR TITLE
Fixed compiler issues under Visual Studio 2015

### PR DESCRIPTION
**Pull request by stunney**
Travis build even passes.

Tested locally against visuals studio 2012, 2013, 2015 as well.

### CODE REVIEW

**Comment by mnunberg** [comment]
`inttypes` is a c99 header which isn't available on Visual Studio 2008. Do you get warnings, or does the upstream version fail to compile.

Also, a message from our lawyers:


To ensure quality review, Couchbase employs a code review system based on Gerrit to manage the workflow of changes in addition to tracking our contributor agreements.

To get this change in and collaborate in code review, please register on Gerrit and accept our CLA. More detailed instructions are available here: https://developer.couchbase.com/open-source-projects#how-to-contribute-code.

**Comment by stunney** [comment]
I changed it to #ifNdef as well, so those includes are only processed during *nix builds.  Did I miss changing one of them?

**Comment by stunney** [comment]
Oh, and I've signed the agreement.

File: .travis.yml [hunk 1]
```diff
@@ -16,4 +16,4 @@ script: |
 
 notifications:
   email:
-    - mark.nunberg@couchbase.com
+    - stephen.tunney@gmail.com
```
Comment (mnunberg) [inline .travis.yml]: and this.. probably a typo!

Comment (stunney) [inline .travis.yml]: Funny. I committed this after creating the PR so my Travis build for my fork would report to me ;)

**Comment by mnunberg** [comment]
you are correct. I just overlooked it. I'll try this and apply it manually against VS 2008.

Regarding the CLA and gerrit: even if you've signed the agreement, it would still need to be merged via gerrit.

**Comment by mnunberg** [comment]
Compiling with clang:

```
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:54:29: warning: length modifier 'I64' results in undefined behavior or no effect with 'x' conversion specifier [-Wformat]
    fprintf(stderr, "CAS=0x%I64x\n", resp->cas);
                           ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:57:45: warning: length modifier 'I64' results in undefined behavior or no effect with 'u' conversion specifier [-Wformat]
        fprintf(stderr, "%-20sSYNCTOKEN=%u,%I64u,%I64u\n",
                                           ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:57:51: warning: length modifier 'I64' results in undefined behavior or no effect with 'u' conversion specifier [-Wformat]
        fprintf(stderr, "%-20sSYNCTOKEN=%u,%I64u,%I64u\n",
                                                 ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:68:39: warning: length modifier 'I64' results in undefined behavior or no effect with 'x' conversion specifier [-Wformat]
        fprintf(stderr, "%-20s CAS=0x%I64x, Flags=0x%x. Size=%lu\n",
                                     ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:144:45: warning: length modifier 'I64' results in undefined behavior or no effect with 'x' conversion specifier [-Wformat]
            "%-20s [%s] Status=0x%x, CAS=0x%I64x\n", key.c_str(),
                                           ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:170:35: warning: length modifier 'I64' results in undefined behavior or no effect with 'x' conversion specifier [-Wformat]
    fprintf(stderr, "[%d] UUID=0x%I64x, Cache=%I64u, Disk=%I64u",
                                 ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:170:48: warning: length modifier 'I64' results in undefined behavior or no effect with 'u' conversion specifier [-Wformat]
    fprintf(stderr, "[%d] UUID=0x%I64x, Cache=%I64u, Disk=%I64u",
                                              ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:170:60: warning: length modifier 'I64' results in undefined behavior or no effect with 'u' conversion specifier [-Wformat]
    fprintf(stderr, "[%d] UUID=0x%I64x, Cache=%I64u, Disk=%I64u",
                                                          ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:174:51: warning: length modifier 'I64' results in undefined behavior or no effect with 'x' conversion specifier [-Wformat]
        fprintf(stderr, "    FAILOVER. New: UUID=%I64x, Cache=%I64u, Disk=%I64u",
                                                 ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:174:64: warning: length modifier 'I64' results in undefined behavior or no effect with 'u' conversion specifier [-Wformat]
        fprintf(stderr, "    FAILOVER. New: UUID=%I64x, Cache=%I64u, Disk=%I64u",
                                                              ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:174:76: warning: length modifier 'I64' results in undefined behavior or no effect with 'u' conversion specifier [-Wformat]
        fprintf(stderr, "    FAILOVER. New: UUID=%I64x, Cache=%I64u, Disk=%I64u",
                                                                          ~^~~~
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:243:41: warning: length modifier 'I64' results in undefined behavior or no effect with 'u' conversion specifier [-Wformat]
        sprintf(buf, "Current value is %I64u.", resp->value);
                                       ~^~~~
/Users/mnunberg/Source/libcouchbase/tests/LOCAL/fts.cc:14:71: warning: format specifies type 'char *' but the argument has type 'lcb_SIZE' (aka 'unsigned long') [-Wformat]
            printf("HTTP Response: %.*s\n", (int)resp->htresp->nbody, resp->htresp->nbody);
                                   ~~~~                               ^~~~~~~~~~~~~~~~~~~
                                   %.*zu
/Users/mnunberg/Source/libcouchbase/tools/cbc.cc:716:45: warning: length modifier 'I64' results in undefined behavior or no effect with 'x' conversion specifier [-Wformat]
        rv = sscanf(args[ii+1].c_str(), "0x%I64x", &cas);
                                           ~^~~~
```

**Comment by mnunberg** [comment]
(and I know I am probably guilty of ignoring the many warnings on Windows boxes.. but the I64x formats are non-standard, AFAIK..)

**Comment by stunney** [comment]
Ah, so when it comes to *nix, something different gets shoved in those MACRO definitions during the cmake -G step?

**Comment by mnunberg** [comment]
It's more that `inttypes.h` itself contains those macros. On *nix it uses the macro definitions in the system itself; whereas on Windows it uses a ported `inttypes.h` called `win_stdin` (https://github.com/couchbase/libcouchbase/blob/master/contrib/win32-defs/win_stdint.h) and `my_inttypes` (https://github.com/couchbase/libcouchbase/blob/master/tools/common/my_inttypes.h)

**Comment by stunney** [comment]
Alright.  Good to know.  How would you recommend I approach this issue?  It seems as though the CPP compiler under VS2015 does not like inline macros in the style of
printf("this is a test of "MACRONAME" your strength") where the macro is clearly outsite of quotes and is defined as 
#define MACRONAME "awesome"

Would it be better to put the macro back in and
string concatenate? stringstream?

Or is it ok as-is?

**Comment by mnunberg** [comment]
What kind of error message does thee CPP give? it should work perfectly fine as it is part of the C standard.. I think it just wants a _space_ between the `"` and `MACRONAME` so instead of `"foo"MACRO` it wants `"foo".

It works fine under 2013; and I don't have easy access to 2015.

**Comment by stunney** [comment]
Just did a clean git clone, cmake -G "Visual Studio 14 2015 Win64" and cmake --build .\ --config Release a log file.  15 errors. File attached but here is the jist:

"C:\Users\Administrator\Desktop\libcouchbase\build\ALL_BUILD.vcxproj" (default target) (1) ->
"C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj" (default target) (3) ->
(ClCompile target) -> 
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(48): error C3688: invalid literal suffix 'PRIx64'; literal operator or literal operator template 'operator ""PRIx64' not found [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(48): error C2664: 'int fprintf(FILE *const ,const char *const ,...)': cannot convert argument 2 from 'const lcb_CAS' to 'const char *const ' [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(51): error C3688: invalid literal suffix 'PRIu64'; literal operator or literal operator template 'operator ""PRIu64' not found [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(62): error C3688: invalid literal suffix 'PRIx64'; literal operator or literal operator template 'operator ""PRIx64' not found [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(138): error C3688: invalid literal suffix 'PRIx64'; literal operator or literal operator template 'operator ""PRIx64' not found [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(164): error C3680: cannot concatenate user-defined string literals with mismatched literal suffix identifiers [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(164): error C3688: invalid literal suffix 'PRIx64'; literal operator or literal operator template 'operator ""PRIx64' not found [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(165): error C2664: 'int fprintf(FILE *const ,const char *const ,...)': cannot convert argument 2 from 'int' to 'const char *const ' [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(168): error C3680: cannot concatenate user-defined string literals with mismatched literal suffix identifiers [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(168): error C3688: invalid literal suffix 'PRIx64'; literal operator or literal operator template 'operator ""PRIx64' not found [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(169): error C2664: 'int fprintf(FILE *const ,const char *const ,...)': cannot convert argument 2 from 'const lcb_U64' to 'const char *const ' [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(237): error C3688: invalid literal suffix 'PRIu64'; literal operator or literal operator template 'operator ""PRIu64' not found [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(237): error C2664: 'int sprintf(char *const ,const char *const ,...)': cannot convert argument 2 from 'const lcb_U64' to 'const char *const ' [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(710): error C3688: invalid literal suffix 'PRIx64'; literal operator or literal operator template 'operator ""PRIx64' not found [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
  C:\Users\Administrator\Desktop\libcouchbase\tools\cbc.cc(710): error C2664: 'int sscanf(const char *const ,const char *const ,...)': cannot convert argument 2 from 'lcb_CAS *' to 'const char *const ' [C:\Users\Administrator\Desktop\libcouchbase\build\tools\cbc.vcxproj]
[build.zip](https://github.com/couchbase/libcouchbase/files/228534/build.zip)

**Comment by stunney** [comment]
I'm trying to think of other tactics to fix this.

Would a printf within the original printf work perhaps?  I'm happy to help in any way that I can.

**Comment by mnunberg** [comment]
Did you attempt placing a space between the macro and the actual literal as above? It seems I'll need to get my hands on a VS 2015 compiler somehow..

**Comment by stunney** [comment]
Success!!!  Putting spaces in seems to have fixed the issue.  I will close this PR and create a new one.

Thanks @mnunberg

**Pull request closed without merging**